File: 프로그래머스/Lv.0/181927. 마지막 두 원소/마지막 두 원소.java
import java.util.Arrays;

class Solution {
    public int[] solution(int[] num_list) {
        int len = num_list.length;

        int[] answer = Arrays.copyOf(num_list, len + 1);

        int last = num_list[len - 1];
        int beforeLast = num_list[len - 2];

        if (last > beforeLast) {
            answer[len] = last - beforeLast;
        } else {
            answer[len] = last * 2;
        }

        return answer;
    }
}

File: 프로그래머스/Lv.0/181927. 마지막 두 원소/마지막 두 원소.kt
class Solution {
    fun solution(num_list: IntArray): IntArray {
        val len = num_list.size

        val answer = num_list.copyOf(len + 1)

        val last = num_list[len - 1]
        val beforeLast = num_list[len - 2]

        if (last > beforeLast) {
            answer[len] = last - beforeLast
        } else {
            answer[len] = last * 2
        }

        return answer
    }
}